File: hapibee-backend/src/main/java/isep/engreq/hapibeebackend/dtos/CreateApiaryDto.java
package isep.engreq.hapibeebackend.dtos;

import isep.engreq.hapibeebackend.models.common.GeoLocation;
import lombok.Data;

import java.util.List;

@Data
public class CreateApiaryDto {

    private String name;
    private String tenantId;
    private List<CreateBeehiveDto> beehives;
    private String town; //"freguesia"
    private GeoLocation geoLocation;
    private String nameOfLocal; //"nome do lugar"
    private boolean intensiveCultivation;
    private boolean inControlledZone;

}

File: happibee-mobile/app/schemas/dtos/CreateApiaryDto.ts
import {BeehiveStructureType, IBeehive} from "../interfaces/IApiary";
import {formatDateToYYYYMMDD} from "../../utils/DateUtils";

export interface CreateApiaryDto {
    name: string;
    tenantId: string;
    beehives: CreateBeehiveDto[];
    town: string;
    geoLocation: GeoLocation;
    nameOfLocal: string;
    intensiveCultivation: boolean;
    inControlledZone: boolean;
}

export interface GeoLocation {
    latitude: number;
    longitude: number;
}

export interface CreateBeehiveDto {
    name: string;
    type: string;
    structureType: BeehiveStructureType;
    color: string;
    source: string;
    frames: number;
    queenRace: string;
    queenAcceptAt: string;
    hiveBodiesOfCreation: number;
    hiveBodiesOfHoney: number;
}

export const mapBeehiveToDto = (beehive: IBeehive): CreateBeehiveDto => {
    return {
        name: beehive.name,
        type: beehive.type,
        structureType: beehive.structureType,
        color: beehive.color,
        source: beehive.source,
        frames: beehive.frames,
        queenRace: beehive.queenRace,
        queenAcceptAt: beehive.queenAcceptAt.toISOString(),
        hiveBodiesOfCreation: beehive.hiveBodiesOfCreation,
        hiveBodiesOfHoney: beehive.hiveBodiesOfHoney,
    };
};
